### PR TITLE
Query tree unit test (partial) cleanup

### PR DESCRIPTION
**Pull request by dalg24**
Incremental step to cleanup the unit test

* Renamed query tree executable
* Separated out spatial and nearest predicates in unit tests
* Updated more tests to use the tree traits

### CODE REVIEW

File: test/tstQueryTreeDegenerate.cpp [hunk 1]
```diff
@@ -137,6 +155,33 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(single_leaf_tree, TreeTypeTraits,
...
+  // tree has a single leaf (unit box)
+  auto const tree =
+      make<Tree>(ExecutionSpace{}, {
+                                       {{{0., 0., 0.}}, {{1., 1., 1.}}},
+                                   });
+
+  BOOST_TEST(!tree.empty());
+  BOOST_TEST(tree.size() == 1);
+  BOOST_TEST(
+      ArborX::Details::equals(tree.bounds(), {{{0., 0., 0.}}, {{1., 1., 1.}}}));
```
Comment (aprokop) [inline test/tstQueryTreeDegenerate.cpp:176]: Have you considered delegating empty/one leaf/two leaves tree construction into a separate function in the file to be called by both tests in order to reduce code duplication?

File: test/tstQueryTreeDegenerate.cpp [hunk 2]
```diff
@@ -272,21 +341,44 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(not_exceeding_stack_capacity, TreeTypeTraits,
...
+  int const n = 4096; // exceed stack capacity which is 64
+  boxes.reserve(n);
+  for (int i = 0; i < n; ++i)
+  {
+    double const a = i;
+    double const b = i + 1;
+    boxes.push_back({{{a, a, a}}, {{b, b, b}}});
+  }
+  auto const bvh = make<ArborX::BVH<typename DeviceType::memory_space>>(
+      ExecutionSpace{}, boxes);
```
Comment (aprokop) [inline test/tstQueryTreeDegenerate.cpp:373]: This is a lot of code duplication. And `bvh`.

**Review by aprokop — COMMENTED** [review]
The current draft seems to introduce a significant amount of duplicated code associated with splitting some test into spatial and nearest. Would it be possible to create a function that does the common setup for both to address that?

**Review by aprokop — DISMISSED** [review]
Well, OK, duplication of code sucks, but not a deal breaker. I'm approving the overall approach.

**Comment by aprokop** [comment]
@dalg24 Maybe remove the spatial/nearest split commit, and merge everything else?

**Comment by dalg24** [comment]
> @dalg24 Maybe remove the spatial/nearest split commit, and merge everything else?

I need a bit more time

**Review by aprokop — DISMISSED** [review]
Can be merged as it is, `intersects(Polytope, Sphere)` is not happening any time soon, and thus separation of nearest and spatial tests is still necessary

**Comment by aprokop** [comment]
HIP failed in the usual manner, ignoring.

**Review by aprokop — APPROVED** [review]

File: test/tstQueryTreeDegenerate.cpp [hunk 3]
```diff
@@ -272,21 +341,44 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(not_exceeding_stack_capacity, TreeTypeTraits,
 
   Kokkos::View<int *, DeviceType> indices("indices", 0);
   Kokkos::View<int *, DeviceType> offset("offset", 0);
-  // query number of nearest neighbors that exceed
-  // capacity of the stack is not a problem
+  // spatial query that find all indexable in the
+  // tree is fine
```
[the remark this answers is not in the document]
Comment (dalg24) [reply test/tstQueryTreeDegenerate.cpp:345]: Original comment was
```C++
  // spatial query that find all indexable in the
  // tree is also fine
```

File: test/tstQueryTreeDegenerate.cpp [hunk 4]
```diff
@@ -216,8 +285,8 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(couple_leaves_tree, TreeTypeTraits,
                          make_reference_solution<int>({0, 1, 0, 1}, {0, 2, 4}));
 }
 
-BOOST_AUTO_TEST_CASE_TEMPLATE(duplicated_leaves, TreeTypeTraits,
-                              TreeTypeTraitsList)
+BOOST_AUTO_TEST_CASE_TEMPLATE(duplicated_leaves_spatial_predicate,
```
[the remark this answers is not in the document]
Comment (dalg24) [reply test/tstQueryTreeDegenerate.cpp:288]: No

File: test/tstQueryTreeDegenerate.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (aprokop) [reply test/tstQueryTreeDegenerate.cpp:345]: I guess it showed up in a different diff for me. I meant, this version:
```
// query number of nearest neighbors that exceed
// capacity of the stack is not a problem
```
I would rephrase it to
```
// perform a query with a number of nearest neighbors chosen to exceed the stack capacity,
// demonstrating that it is not a problem
```

[the remark this answers is not in the document]
Comment (dalg24) [reply test/tstQueryTreeDegenerate.cpp:345]: I would rather not deal with it now and merge as is

[the remark this answers is not in the document]
Comment (aprokop) [reply test/tstQueryTreeDegenerate.cpp:345]: Just fix it.
